subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/release/tasks/ReleaseNeeded.java

```diff
@@ -15,43 +16,57 @@
+        String message = pair.getRight();
+
+        if (!releaseNeeded && task.isExplosive()) {
+            throw new GradleException(message);
+        } else {
+            LOG.lifecycle(message);
+        }
+        return releaseNeeded;
     }
 
-    public boolean releaseNeeded(ReleaseNeededTask task, EnvVariables envVariables) {
+    Pair<Boolean, String> releaseNeeded(ReleaseNeededTask task, EnvVariables envVariables) {
         boolean skipEnvVariable = envVariables.getNonEmptyEnv(SKIP_RELEASE_ENV) != null;
         LOG.lifecycle("  Environment variable {} present: {}", SKIP_RELEASE_ENV, skipEnvVariable);
 
         boolean commitMessageEmpty = task.getCommitMessage() == null || task.getCommitMessage().trim().isEmpty();
         boolean skippedByCommitMessage = !commitMessageEmpty && task.getCommitMessage().contains(SKIP_RELEASE_KEYWORD);
-        LOG.lifecycle("  Commit message to inspect for keyword '{}': {}",
-                SKIP_RELEASE_KEYWORD,
-                commitMessageEmpty ? "<unknown commit message>" : "\n" + task.getCommitMessage());
+        boolean forcedByCommitMessage = !commitMessageEmpty && task.getCommitMessage().contains(FORCE_RELEASE_KEYWORD);
+        LOG.lifecycle("  Commit message to inspect for keywords '{}' and '{}': {}",
+            SKIP_RELEASE_KEYWORD, FORCE_RELEASE_KEYWORD,
+            commitMessageEmpty ? "<unknown commit message>" : "\n" + task.getCommitMessage());
 
         boolean releasableBranch = task.getBranch() != null && task.getBranch().matches(task.getReleasableBranchRegex());
         LOG.lifecycle("  Current branch '{}' matches '{}': {}", task.getBranch(), task.getReleasableBranchRegex(), releasableBranch);
 
-        ComparisonResults results = new ComparisonResults(task.getComparisonResults());
-        boolean publicationsIdentical = results.areResultsIdentical();
-        LOG.lifecycle(results.getDescription());
+        if (releasableBranch) {
+            if (skippedByCommitMessage) {
+                return Pair.of(false, " Skipping release due to skip release keyword in commit message.");
+            } else if (skipEnvVariable) {
+                return Pair.of(false, " Skipping release due to skip release env variable.");
+            } else if (task.isPullRequest()) {
+                return Pair.of(false, " Skipping release due to is PR.");
+            } else if (forcedByCommitMessage) {
+                return Pair.of(true, " Releasing due to force release keyword in commit message.");
+            } else {
+                ComparisonResults results = new ComparisonResults(task.getComparisonResults());
+                boolean publicationsIdentical = results.areResultsIdentical();
+                LOG.lifecycle(results.getDescription());
 
-        boolean releaseNotNeeded = publicationsIdentical || skipEnvVariable || skippedByCommitMessage || task.isPullRequest() || !releasableBranch;
-
-        String message = "  Release is needed: " + !releaseNotNeeded +
```

Comment:
This message is useful for debugging and I don't think we want to get rid of it. See ticket #209 for how we can improve this message.